test/test_torch.py

```diff
@@ -12340,6 +12340,172 @@ def test_against_np(tensor, bins=100, min=0, max=0):
+        # scalar input and 1 bin -- should return a 1-dimensional tensor, not a scalar.
+        actual = torch.histogram(
+            torch.tensor(0, dtype=torch.float, device=device),
+            bins=1, range=(0, 3))[0]
+        self.assertEqual(
+            torch.tensor([1], dtype=torch.int64, device=device),
+            actual)
+        # tensors with inf; range not provided -- should throw a RuntimeError
+        with self.assertRaisesRegex(RuntimeError, r'range of \[inf, inf\] is not finite'):
+            torch.histogram(torch.tensor([float("inf")], dtype=torch.float, device=device))
+        with self.assertRaisesRegex(RuntimeError, r'range of \[1, inf\] is not finite'):
+            torch.histogram(torch.tensor([1., 2., float("inf")], dtype=torch.float, device=device))
+        # tensors with inf; range provided
+        self.assertEqual(
+            torch.histogram(torch.tensor([float("inf")], dtype=torch.float, device=device),
+                            bins=1, range=(0, 3))[0],
+            torch.tensor([0], dtype=torch.long, device=device))
+        self.assertEqual(
+            torch.histogram(torch.tensor([1., 2., float("inf")], dtype=torch.float, device=device),
+                            bins=4, range=(0, 3))[0],
+            torch.tensor([0, 1, 1, 0], dtype=torch.long, device=device))
+        # tensor with nan -- should throw a RuntimeError
+        with self.assertRaisesRegex(RuntimeError, r'range of \[nan, nan\] is not finite'):
+            torch.histogram(torch.tensor([float("nan")], dtype=torch.float, device=device))
+        # tensors with min > max -- should throw a RuntimeError
+        with self.assertRaisesRegex(RuntimeError, "max must be larger than min"):
+            torch.histogram(torch.tensor([1., 2., 3.], dtype=torch.float, device=device),
+                            bins=4, range=(5, 1))
+
+        # test against numpy.histogram()
+        def test_against_np_uniformbins(tensor, bins=100, min=0, max=0, weights=None, density=False):
+            if min == 0 and max == 0:
+                min = tensor.min().item()
+                max = tensor.max().item()
+            nparr = tensor.cpu().numpy()
+            if weights is not None:
+                npweights = weights.cpu().numpy()
+            else:
+                npweights = None
+            actual = torch.histogram(tensor, bins=bins, weights=weights, range=(min, max), density=density)
+            expected = np.histogram(nparr, bins=bins, range=(min, max), weights=npweights, density=density)
+            self.assertEqual(actual[0], torch.from_numpy(expected[0]))
+            if torch.is_floating_point(tensor):
+                self.assertEqual(actual[1], torch.from_numpy(expected[1]))
+            else:
+                self.assertEqual(actual[1], torch.from_numpy(expected[1]).to(torch.get_default_dtype()))
+
+        def test_against_np_custombins(tensor, bins, weights=None, density=False, rtol=None, atol=None):
```

Comment:
This should also be a helper function in a new test called "test_histogram_vs_np_custom_bins" to improve test granularity and readability.